peps/pep-0735.rst

```diff
@@ -0,0 +1,613 @@
+
+This spec does not allow for such a structure. There is only one way to write
+down a PEP 508 dependency, as a string. This has two positive effects:
+
+* There are fewer ways of writing identical data
+
+* It is not possible, by design, to combine a PEP 508 specifier with other
+  key-value pairs in a dependency specifier
+
+Specification
+=============
+
+This PEP defines a new section (table) in ``pyproject.toml`` files named
+``dependency-groups``. The ``dependency-groups`` table contains an arbitrary
+number of user-defined keys, each of which has, as its value, a list of
+requirements specifiers (defined below).  These keys must match the following
+regular expression: ``[a-z0-9][a-z0-9-]*[a-z0-9]``. Meaning that they must be
+all lower-case alphanumerics, with ``-`` allowed only in the middle, and at
+least two characters long. These requirements are chosen so that the
+normalization rules used for PyPI package names are unnecessary as the names
+are already normalized.
+
+Requirements specifiers will use a definition based on standardized
+`Dependency Specifiers <https://packaging.python.org/en/latest/specifications/dependency-specifiers/>`__
+introduced in :pep:`508`.
+This PEP also proposes an object format to define a dependency, called a "PEP
+735 Dependency" (defined below). The elements in ``[dependency-groups]``
+lists must either be strings, in which case they must be valid Dependency
+Specifiers or else PEP 735 Dependencies.
+
+PEP 735 Dependencies
+--------------------
+
+PEP 735 Dependencies are objects which define a set of dependencies. They do
+not necessarily refer to a single package, as will become clear from the
+definitions below.
+
+There are two forms of PEP 735 Dependencies: "Dependency Group Includes" and
+"Path Dependencies".
+
+Dependency Group Include
+''''''''''''''''''''''''
+
+A Dependency Group Include includes the dependencies of another Dependency
+Group in the current Dependency Group.
+
+An include is defined as an object with exactly one key, ``"include"``, whose
+value is a string, the name of another Dependency Group.
+
+For example, ``{include = "test"}`` is an include which expands to the
+contents of the ``test`` Dependency Group.
+
+Path Dependency
+'''''''''''''''
+
+A Path Dependency is a dependency on a package found via a local filesystem
+path OR on dependencies of a package found via a local filesystem path.
+
+It contains the following keys, with associated types:
+
+* ``path``: a string, required
+* ``extras``: a list of strings, optional
+* ``editable``: a boolean, optional
+* ``only_deps``: a boolean, optional
+
+For a simple example, ``{path = ".", editable = true, extras = ["mysql"]}`` is
+a Path Dependency on the current project including its ``mysql`` extra. In
+``requirements.txt`` files, a similar idea can be expressed as ``-e .[mysql]``.
+
+``path``
+~~~~~~~~
+
+The ``path`` must refer to the path to a built distribution (wheel, sdist, or
```

Comment:
I assume this path is relative to the directory that the current `pyproject.toml` is in. That should be made explicit.